docs/mqtt.md

```diff
@@ -1060,6 +1061,121 @@ A sample motion sensor topic and payload configuration is:
+inherited from the battery sensor inputs.  The hidden door sensor adds
+another possible state change the 2 groups configuration where it will report
+open as ON on group 0x01 and closed as ON on group 0x02.  The raw Insteon
+reported battery voltage level, the low battery threshold raw Insteon level
+setting and the heart beat interval are all reported over MQTT.
+
+The following variables are available for templates:
+
+   'batt_volt' is the raw Insteon voltage level
+   'low_batt_volt' is the raw Insteon voltage level to trigger low bat
+   'hb_interval' is the interval between heart beats sent by device.
+   Interval = 5 mins x this setting.  0 = 24 hours
+
+
+A sample hidden door sensor topic and payload configuration is:
+
+   ```
+   hidden_door:
+     battery_voltage_topic: 'insteon/{{address}}/battery/state'
+     battery_voltage_payload: '{"voltage" : {{batt_volt}}}'
+
+     low_battery_voltage_topic: 'insteon/{{address}}/low_battery/state'
+     low_battery_voltage_payload: '{"low_voltage_threshold" : {{low_batt_volt}}}'
+
+     heart_beat_interval_topic: 'insteon/{{address}}/heartbeat/state'
+     heart_beat_interval_payload: '{"heart_beat_interval" : {{hb_interval}}}'
+   ```
+
+To set configuration option on the device, first press and hold the device link
+button until it beeps and the LED starts flashing.  Next tell insteon-mqtt the
+device is awake with:
+
+   ```
+   { "cmd" : "awake" }
+   ```
+
+This will tell insteon-mqtt to send the command right away rather than
+queuing until the deice is awake.
+
+View Current configuration in log:
+
+   ```
+   { "cmd" : "get_flags" }
+   ```
+
+This configuration can be changed with the following command:
+
+   ```
+   { "cmd" : "set_flags", "key" : value }
+
+   ```
+
+An example to turn on two groups:
+
+   ```
+   { "cmd" : "set_flags", "two_groups" : 1 }'
+   ```
+
+Configuration is available for the following options:
+
+The following key/value pairs are available:
+
+   - cleanup_report = 1/0: tell the device whether or not to send cleanup
+   reports
+
+   - led_disable = 1/0: disables small led on back of device to blink on
+   state change
+
+   - link_to_all = 1/0: links to 0xFF group (all available groups)
+
+   - two_groups = 1/0: Report open/close on group 1 or report open on group 1
+   and closed on 2
+
+   - prog_lock = 1/0: prevents device from being programmed by local button
+   presses
+
+   - repeat_closed = 1/0: Repeat open command every 5 mins for 50 mins
+
+   - repeat_open = 1/0: Repeat open command every 5 mins for 50 mins
+
+   - stay_awake = 1/0: keeps device awake - but uses a lot of battery
+
+Beyond these flags there are two additional settings:
+
+   - Low Battery threshold.  This is the raw Insteon voltage level that where
+   the device will trigger a group 0x03 low battery warning. Example to set to
+   64 below:
+   
+   ```
+   { "cmd" : "set_low_battery_voltage", "voltage" : 64 }
+   ```
+
```

Comment:
Do you have any idea about the voltage conversion to the raw values here?  At minimum it would be nice to tell the user what that calculation is.  It might also be nice to accept a nice float of a value of voltage and convert that to a raw byte for the user.

Reply:
The dev notes doc gives some clues.  The state these points:

61=~1.75V 
54=~1.6V
51=~1.5V
40=~1.25V (default low battery mark)

(Note these are HEX raw values)

They note that these are approx values only.  

I put together a spreadsheet and these values don't equate to something linear, so I've never bothered trying to translate from the raw number to a voltage.  To be honest I'm not sure it really matters what the voltage is.  I just want to know when the level is approaching what I typically see as my warning level so I can take action.  Whether that is presented as 40 HEX, 64 DEC or 1.25V, I don't really care.  I put the sheet here if you want to see:

https://docs.google.com/spreadsheets/d/18M4HsAcQHARZTWSjM2e51YdY5MXqat1uqpXT1fBnC8I/edit?usp=sharing

Reply:
Fair enough, can you add the few points that you have to the documentation.  Many users are not super sophisticated and this would cut down on support requests.

Reply:
Thanks, I will add this to the docs.

Reply:
Complete.